Assets/ObstacleTower/Scripts/AgentLogic/ObstacleTowerAgent.cs

```diff
@@ -294,11 +328,11 @@ public override void AgentAction(float[] vectorAction, string textAction)
 
         _collisions.Clear();
 
-        if (!IsDone())
-        {
-            CheckOutOfBounds();
-            CheckTimeout();
-        }
+        // if (!IsDone())
```

Comment:
Should this be commented out? I assume this is because being done is no longer possible.

These checks should still be happening.

Reply:
Curious to take you opinion here. These checks are now impossible since the agent is never in a `done` state anymore (the agent resets immediately). Why do we need to check if the Agent is done here ?

Reply:
Well it was supposed to run when it was not done, so I think we can just run it at all times now.

Reply:
Good call, I re-added them.
I wonder if there will be issues when starting a new episode with OnColisionEnter 
How did you test the previous releases? Is there a workflow ?

Reply:
There was not an automated testing method. Running 1000 steps with the gym wrapper and testing the environment in player mode in the editor to ensure that that each of the floors spawn correctly is typically what was done.